File: fixtures/scale/java/medium/src/calc/lexer/Token.java
package calc.lexer;

public class Token {
    private final TokenType type;
    private final String text;
    private final int position;

    public Token(TokenType type, String text, int position) {
        this.type = type;
        this.text = text;
        this.position = position;
    }

    public TokenType getType() {
        return type;
    }

    public String getText() {
        return text;
    }

    public int getPosition() {
        return position;
    }

    public double numberValue() {
        return Double.parseDouble(text);
    }
}

File: fixtures/scale/dart/medium/lib/lexer/token.dart
import 'token_type.dart';

class Token {
  final TokenType type;
  final String text;
  final int position;

  Token(this.type, this.text, this.position);

  double numberValue() => double.parse(text);

  @override
  String toString() => '$type($text)';
}
